url75694980d5_fabioalmeid_tp01ftc_tgz-pJ8-agentes_jade_Centralizador_ExemploParserGramaticaCentralizadorJ8/src/agentes/jade/Centralizador/AgenteCentralizador.java

```diff
@@ -0,0 +1,415 @@
+											StringBuilder temp = new StringBuilder();
+											temp.append(getLocalName() + ": Pressao arterial ruim, Sist -> " + String.valueOf(af.getQuantidade1()) + 
+													" Diast -> " + String.valueOf(af.getQuantidade2()));
+											temp.append(getReferenceValue("pressao"));
+											temp.append(". Aplicando remedio.");
+											System.out.println(temp.toString());
+											updateOutputFile(temp.toString(), file);
+											
+											TarefaCentralizador tarefa = new TarefaCentralizador();
+											tarefa.setAcao(new EApply1());
+											Medicamento medicamento = new Medicamento();
+											medicamento.setQuantidade(20);
+											medicamento.setRemedio(new ERemedy2());
+											tarefa.setMedicacao(medicamento);
+											sendMessageToAgent( tarefa.prettyPrinterTarefa(), "atuador" + indexCurrentAgent);
+											atuadorIsBusy[Integer.parseInt(indexCurrentAgent)].setApplyParacetamol(true);
+										}
+									}
+
+								}
+							}
+						} catch (Exception e) {
+							System.out.println(getLocalName() + ". Erro ao ler resposta do monitor: " + e.getMessage());
+						}
+					} else if (atuador.contains(sender)){
+						System.out.println(getLocalName() + ": Msg recebida do " + msg.getSender().getLocalName() + " -->" + msg.getContent());
+						updateOutputFile(getLocalName() + ": Msg recebida do " + msg.getSender().getLocalName() + " -->" + msg.getContent(), file);
+					}
+						
+					else {// ERRO
+						System.out.println(getLocalName() + ": "
+								+ sender.getLocalName() + ", "
+								+ msg.getContent());
+					}
+					// System.out.println("**********************");
+					// System.out.println();
+				}
+
+			}
+		});
+	}
+
+	private AID CreatNewAgent(AgentContainer container, String agentName, String agentAdrress) {
+		AID minhaID = null;
+		try {
+			System.out.println(getLocalName() + ": Agente " + agentName + " criado com sucesso");
+			container.createNewAgent(agentName, agentAdrress, null).start();
+		} catch (StaleProxyException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
+		}
+		return minhaID;
+	}
+
+	private void sendMessageToAgent(String mensagem, String agentName) {
+		ACLMessage message = new ACLMessage(ACLMessage.REQUEST);
+		AID receiver = new AID(agentName, AID.ISLOCALNAME);
+		message.addReceiver(receiver);
+		message.setContent(mensagem);
+		System.out.println(getLocalName() + ": " + receiver.getLocalName() + ", " + message.getContent());
+		updateOutputFile(getLocalName() + ": " + receiver.getLocalName() + ", " + message.getContent(), file);
+		//System.out.println(getLocalName() + "para " + agentName + " : " + mensagem);
+		send(message);
+	}
+	
+	private String getReferenceValue(String tipo){
+		StringBuilder result = new StringBuilder();
+		result.append(". Valor de referencia: ");
+		if (tipo == "temperatura")
+			result.append("35<T<" + String.valueOf(INICIO_FEBRE));
+		else if (tipo == "hemoglobina")
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.